src/Processors/Formats/Impl/ParquetBlockInputFormat.cpp

```diff
@@ -238,6 +241,50 @@ static std::optional<Field> decodePlainParquetValueSlow(const std::string & data
     return field;
 }
 
+static std::size_t getHeaderIndexByColumnName(const std::string & column_name, const Block & header)
```

Comment:
> You're mapping index to name (in findRequiredIndices()), then name back to index (here). Consider propagating the index instead. The name doesn't seem to be used for anything else, so maybe it can just be replaced with index?

I am still not sure if I can simply drop the names, but that might be due to my lack of knowledge on KeyCondition class. KeyCondition (and my former "ParquetBloomFilterCondition" class) have access nodes. In the case of `KeyCondition`, each node has a `key_column` attribute that refers to the index of that column in the key_column_names, which does not match the parquet file indexes.

In order to get the bloom filter readers out of the parquet file, I need to provide the parquet column index. Upon traversing the RPN, I need to grab the right bloom filter based on the node index. How do I glue these two together?

The obvious way that comes to mind is a mapping between name and file index or header index and column index. I think both would work.

> Seems that it'll fail for named tuples. 

and

> Seems that it'll behave incorrectly for compound types

I think I'll drop the support for compound types for now. After all, I could not even generate bloom filters for map types.